### DISCUSSION TITLE
Cog source code architecture

### DISCUSSION THREAD

**Discussion by kd-1993** [Igalia/cog#424] [General]
Hi Folks,

I have try to find the code architecture of COG but not able to find it so anyone can help to get this.
Mainly my focus is on APIs used to interact with the Wayland, webkit.

Any help will be appreciated.

**Comment by philn** [comment]
There's a very basic doc about WPE's design in https://wpewebkit.org/about/architecture.html

**Comment by kd-1993** [comment]
Hi @philn 
Thanks for the reply.
I require architecture of COG not a Wpewebkit.

**Reply by philn** [reply]
You've seen this I guess, https://github.com/Igalia/cog/tree/master/docs
You can also read the code :) We welcome any contributions, documentation-related patches as well.

**Reply by aperezdc** [reply]
Also the built documentation can be found at https://igalia.github.io/cog and it's kept up-to-date with the state of the `master` branch automatically.
